### DISCUSSION TITLE
Improve the header navigation feature

### DISCUSSION THREAD

**Discussion by xavier-chiliz** [GitbookIO/community#472] [Feature Requests]
When creating a space or collection customization, one can add links in the header, and even create a menu of links, which is great (but, as far as I can tell, [sparsely documented](https://docs.gitbook.com/published-documentation/customization/space-customization#layout-manage-navigation-options-for-your-content)) as it removes external links from the sidebar.

<img width="536" alt="image" src="https://github.com/GitbookIO/community/assets/134376951/9b9f6447-9421-44c4-83e9-49a5e1a65a7e">

There are however two limitations to this:
1. **The main items must have a link.** 
It would be great if the main item could be linkless, so that we could have a simple menu header with links.
2. **It seems that the number of links in each menu is limited to 4**. Any additional item is just not displayed.
It would be great to expand that limit to 6 or 8.
It would also be nice to properly indicate how many links can be put in each menu.

Also, a **bug**: not putting links in the main items and saving, deletes the whole navigation. Not a great experience.

Thank you all!

**Comment by addisonschultz** [comment]
Hey @xavier-chiliz! Thanks so much for the detailed explanations. I've tested these workflows and also see a few of the broken things you've pointed out. I've made some tickets for our team to look into! 

One thing you mentioned though, should be possible:

> It seems that the number of links in each menu is limited to 4. Any additional item is just not displayed.
> It would be great to expand that limit to 6 or 8.

As of now you can add 6 main header links and 8 sub navigation links - the button to add more links will become disabled when you hit that limit. Can you share more about what you're experiencing? 

See the photo below for a reference!

<img width="1624" alt="Screenshot 2024-02-05 at 13 39 00" src="https://github.com/GitbookIO/community/assets/42930383/96c95f2b-3478-4fad-8fb0-874c072128f1">

**Comment by xavier-chiliz** [comment]
> As of now you can add 6 main header links and 8 sub navigation links - the button to add more links will become disabled when you hit that limit. Can you share more about what you're experiencing?

Indeed, testing right now give me the ability to add more than 4 links. It didn't when I wrote that ticket, so I guess something was fixed

The subnav links are displayed in a menu with a scrollbar rather than getting the full list directly, which is less than ideal but at least it works.
<img width="170" alt="image" src="https://github.com/GitbookIO/community/assets/134376951/caabd8ef-898c-44ab-a467-cd3a1cad124e">
Thanks!
